daft/io/_parquet.py

```diff
@@ -63,8 +66,15 @@ def read_parquet(
 
     pytimeunit = coerce_int96_timestamp_unit._timeunit if coerce_int96_timestamp_unit is not None else None
 
+    if isinstance(path, list) and row_groups is not None and len(path) != len(row_groups):
+        raise ValueError("row_groups must be the same length as the list of paths provided.")
+    if isinstance(row_groups, list) and not isinstance(row_groups[0], list) and isinstance(path, list):
+        raise ValueError("row_groups must match the same structure as the list of paths provided.")
```

Comment:
WDYT if we instead limit `row_groups` to only be allowed if `path` is also a list? Might simplify the logic here and I can't imagine a use-case where a user might want "only rg [0, 1] across all files in this glob".

Reply:
Sorry @universalmind303 i thought about it more and I think you're right from before

We also want to allow the common use-case of users passing in a single file, and a single list of rowgroups in that file